src/components/molecule_3d.jsx

```diff
@@ -156,45 +172,48 @@ class Molecule3d extends React.Component {
         });
       }
 
-      glviewer.setStyle({ serial: atom.serial }, libStyle);
-    });
+      const libStyleString = JSON.stringify(libStyle);
+      stylesByAtom.set(atom.serial, libStyleString);
 
-    // Shapes
-    this.props.shapes.forEach((shape) => {
-      if (shape.type) {
-        glviewer[`add${shape.type}`](libUtils.getShapeSpec(shape, this.setSelectionTrait));
+      // If the style string for this atom is the same as last time, then no
+      // need to set it again
+      if (this.lastStylesByAtom &&
+        this.lastStylesByAtom.get(atom.serial) === libStyleString) {
```

Comment:
is the map just using strings for keys? If so a regular literal object easier? e.g. if (this.lastStylesByAtom[atom.serial] === ...
Maps are slighty verbose unless you actually need them i.e. DOM elements as keys etc.